django/db/models/aggregates.py

```diff
@@ -115,42 +166,51 @@ def get_group_by_cols(self):
         return []
 
     def as_sql(self, compiler, connection, **extra_context):
-        extra_context["distinct"] = "DISTINCT " if self.distinct else ""
-        if self.filter:
-            if connection.features.supports_aggregate_filter_clause:
-                try:
-                    filter_sql, filter_params = self.filter.as_sql(compiler, connection)
-                except FullResultSet:
-                    pass
-                else:
-                    template = self.filter_template % extra_context.get(
-                        "template", self.template
-                    )
-                    sql, params = super().as_sql(
-                        compiler,
-                        connection,
-                        template=template,
-                        filter=filter_sql,
-                        **extra_context,
-                    )
-                    return sql, (*params, *filter_params)
-            else:
+        if (
+            self.distinct
+            and not connection.features.supports_aggregate_distinct_multiple_argument
+            and len(super().get_source_expressions()) > 1
+        ):
+            raise NotSupportedError
```

Comment:
Minor but I think we'll want to raise a more explicit message here as it might be user facing


```suggestion
            raise NotSupportedError(
                f"{self.name} doesn't support `distinct` with multiple expressions on {connection.vendor}"
            )
```

We should also add a `@skipIfDBFeature("supports_aggregate_distinct_multiple_argument")` test for it that `assertRaisesMessage`.